src/PoolManager.sol

```diff
@@ -294,6 +294,14 @@ contract PoolManager is IPoolManager, ProtocolFees, NoDelegateCall, ERC6909Claim
         _accountDelta(currency, paid.toInt128(), msg.sender);
     }
 
+    /// @inheritdoc IPoolManager
+    function clear(Currency currency, uint256 amount) external onlyWhenUnlocked {
+        int256 current = currency.getDelta(msg.sender);
+        int128 amountDelta = amount.toInt128();
+        if (amountDelta != current) revert MustClearExactBalance();
```

Comment:
maybe `MustClearExactPositiveDelta`? So that when its thrown in the 0-case or the negative case it still sounds right